server/src/main/java/org/jboss/as/server/deployment/module/descriptor/DeploymentStructureDescriptorParser.java

```diff
@@ -263,12 +283,19 @@ private void handleDeployment(final DeploymentPhaseContext phaseContext, final D
         for (final String classFileTransformer : rootDeploymentSpecification.getClassFileTransformers()) {
             moduleSpec.addClassFileTransformer(classFileTransformer);
         }
-        //handle annotations
+        // handle annotations
         for (final ModuleIdentifier dependency : rootDeploymentSpecification.getAnnotationModules()) {
-            deploymentUnit.addToAttachmentList(Attachments.ADDITIONAL_ANNOTATION_INDEXES, dependency);
-            //additional modules will not be created till much later, a dep on them would fail
-            if(dependency.getName().startsWith(ServiceModuleLoader.MODULE_PREFIX) && !additionalModules.contains(dependency)) {
-                phaseContext.addToAttachmentList(Attachments.NEXT_PHASE_DEPS, ServiceModuleLoader.moduleServiceName(dependency));
+            ModuleIdentifier identifier = dependency;
+            for (AdditionalModuleSpecification module : additionalModules.values()) {
+                if (module.getAliases().contains(dependency)) {
+                    identifier = module.getModuleIdentifier();
+                    break;
+                }
+            }
+            deploymentUnit.addToAttachmentList(Attachments.ADDITIONAL_ANNOTATION_INDEXES, identifier);
+            // additional modules will not be created till much later, a dep on them would fail
+            if (identifier.getName().startsWith(ServiceModuleLoader.MODULE_PREFIX) && !moduleIncluded(additionalModules.values(), identifier)) {
```

Comment:
do you need to iterate on `additionalModules.values()` to check if it it includes the` identifier`?

Could you just check if `additionalModules.keys()` contains `identifier` instead?

Reply:
Yes, the whole private moduleIncluded method can be eliminated.